apps/omg_watcher/test/block_getter/core_test.exs

```diff
@@ -642,6 +642,130 @@ defmodule OMG.Watcher.BlockGetter.CoreTest do
              {:error, :not_at_block_beginning}
   end
 
+  test "BlockGetter omits submissions of already applied blocks" do
+    state =
+      init_state(synced_height: 1, start_block_number: 1000)
+      |> Core.get_numbers_of_blocks_to_download(5_000)
+      |> assert_check([2_000, 3_000, 4_000])
+      |> handle_downloaded_block(%Block{number: 2_000})
+
+    {_, _, _, %Core{eth_height_done_by_blknum: map}} =
+      Core.get_blocks_to_apply(
+        state,
+        [%{blknum: 1_000, eth_height: 1}, %{blknum: 2_000, eth_height: 2}],
+        2
+      )
+
+    assert %{2_000 => 2} == map
+  end
+
+  test "an unapplied block appears in an already synced eth block (due to reorg)" do
+    state =
+      init_state(synced_height: 2, start_block_number: 1000)
+      |> Core.get_numbers_of_blocks_to_download(5_000)
+      |> assert_check([2_000, 3_000, 4_000])
+      |> handle_downloaded_block(%Block{number: 2_000})
+      |> handle_downloaded_block(%Block{number: 3_000})
+
+    {[{%Block{number: 2_000}, 1}, {%Block{number: 3_000}, 3}], 2, _, _} =
+      Core.get_blocks_to_apply(
+        state,
+        [%{blknum: 2_000, eth_height: 1}, %{blknum: 3_000, eth_height: 3}],
+        3
+      )
+  end
+
+  test "BlockGetter updates mappings of eth_height done by blknum" do
```

Comment:
the title of the test should not delve into the internals (`updates mappings of...`). Test the behavior you're getting (correct synced height being returned most likely) & title accordingly